core/workspace_drag_surface_svg.js

```diff
@@ -0,0 +1,181 @@
+ */
+Blockly.workspaceDragSurfaceSvg.prototype.createDom = function() {
+  if (this.SVG_) {
+    return;  // Already created.
+  }
+
+  /**
+  * FILL dom structure in!
+  */
+  this.SVG_ = Blockly.utils.createSvgElement('svg', {
+    'xmlns': Blockly.SVG_NS,
+    'xmlns:html': Blockly.HTML_NS,
+    'xmlns:xlink': 'http://www.w3.org/1999/xlink',
+    'version': '1.1',
+    'class': 'blocklyWsDragSurface'
+  }, null);
+  this.container_.appendChild(this.SVG_);
+};
+
+
+/**
+ * Translate the entire drag surface during a drag.
+ * We translate the drag surface instead of the blocks inside the surface
+ * so that the browser avoids repainting the SVG.
+ * Because of this, the drag coordinates must be adjusted by scale.
+ * @param {Number} x X translation for the entire surface
+ * @param {Number} y Y translation for the entire surface
+ * @package
+ */
+Blockly.workspaceDragSurfaceSvg.prototype.translateSurface = function(x, y) {
+  // Force values to have two decimal points.
+  // This is a work-around to prevent a bug in Safari, where numbers close to 0
+  // are sometimes reported as something like "2.9842794901924208e-12".
+  // That is incompatible with translate3d, causing bugs.
+  x = x.toFixed(0);
+  y = y.toFixed(0);
+
+  // Ignorning browsers that don't support translate3d at the moment.
+  var transform =
+    'transform: translate3d(' + x + 'px, ' + y + 'px, 0px); display: block;';
+  this.SVG_.setAttribute('style', transform);
+};
+
+/**
+ * Reports the surface translation in scaled workspace coordinates.
+ * Use this when finishing a drag to return blocks to the correct position.
+ * @return {goog.math.Coordinate} Current translation of the surface
+ * @package
+ */
+Blockly.workspaceDragSurfaceSvg.prototype.getSurfaceTranslation = function() {
+  return Blockly.utils.getRelativeXY(this.SVG_);
+};
+
+/**
+ * Clear the  and hide the contents of the  surface;  Move everything back to
+ * newSurface.
+ * @param {!SVGElement} newSurface The element to put the drag surface contents
+ * into.
+ * @package
+ */
+Blockly.workspaceDragSurfaceSvg.prototype.clearAndHide = function(newSurface) {
+  var blockCanvas = this.SVG_.childNodes[0];
+  var bubbleCanvas = this.SVG_.childNodes[1];
+  if (!blockCanvas || !bubbleCanvas ||
+      !blockCanvas.classList.contains('blocklyBlockCanvas') ||
+      !bubbleCanvas.classList.contains('blocklyBubbleCanvas')) {
+    throw 'Couldn\'t clear and hide the drag surface.  A node was missing.';
+  }
+
+  // If there is a previous sibling, put the blockCanvas back right afterwards,
+  // otherwise insert it as the first child node in newSurface.
+  if (this.previousSibling_ != null) {
+    Blockly.utils.insertAfter_(blockCanvas, this.previousSibling_);
+  } else {
+    newSurface.insertBefore(blockCanvas, newSurface.firstChild);
+  }
+
+  // Reattach the bubble canvas after the blockCanvas.
+  Blockly.utils.insertAfter_(bubbleCanvas, blockCanvas);
+  this.SVG_.style.display = 'none';
```

Comment:
Newline, then 
// Hide the drag surface
this.SVG_.style.display = 'none';
...